<?xml version="1.0"?>
<launch>
  <arg name="ros_param_path"   default="$(find-pkg-share bevdet_ros)/config/bevdet.param.yaml"/>
  <arg name="data_param_path"  default="$(find-pkg-share bevdet_ros)/config/bevdet/configure.yaml"/>
  <arg name="model_param_path" default="$(find-pkg-share bevdet_ros)/config/bevdet/cfgs/bevdet.yaml"/>

  <arg name="input/odom" default="/odom"/>
  <arg name="input/cam_front_left" default="/CAM_FRONT_LEFT/image_raw"/>
  <arg name="input/cam_front" default="/CAM_FRONT/image_raw"/>
  <arg name="input/cam_front_right" default="/CAM_FRONT_RIGHT/image_raw"/>
  <arg name="input/cam_back_left" default="/CAM_BACK_LEFT/image_raw"/>
  <arg name="input/cam_back" default="/CAM_BACK/image_raw"/>
  <arg name="input/cam_back_right" default="/CAM_BACK_RIGHT/image_raw"/>

  <arg name="use_sim_time" default="true" description="use_sim_time"/>

  <node name="bevdet_node" pkg="bevdet_ros" exec="bevdet_ros_node" output="screen" >
    <param from="$(var ros_param_path)" allow_substs="true"/>
    <param from="$(var data_param_path)" allow_substs="true"/>
    <param from="$(var model_param_path)" allow_substs="true"/>

    <remap from="~/input/odom" to="$(var input/odom)"/>
    <remap from="~/input/image_fl" to="$(var input/cam_front_left)"/>
    <remap from="~/input/image_f" to="$(var input/cam_front)"/>
    <remap from="~/input/image_fr" to="$(var input/cam_front_right)"/>
    <remap from="~/input/image_bl" to="$(var input/cam_back_left)"/>
    <remap from="~/input/image_b" to="$(var input/cam_back)"/>
    <remap from="~/input/image_br" to="$(var input/cam_back_right)"/>
  </node>
</launch>